### PR TITLE
E2E tests for Sample plugin

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
This pull request implements the following changes:

- Creation of a 'sample-plugin' under the 'test' directory.
- Addition of E2E test cases for the 'sample-plugin' to verify its functionality using the E2E framework API.
- Add GitHub workflow to execute E2E test cases for the 'sample-plugin.'
- CLI E2E framework enhancements to accommodate E2E testing for the 'sample-plugin.'

The goal of this PR is to showcase the capabilities of the Tanzu CLI E2E framework for implementing end-to-end tests for plugin developers.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- We have added a GitHub workflow to execute E2E test cases, which cover both the sample plugin's functionality and the demonstration of E2E framework API usage.
- The results of the sample plugin tests are also presented in the Github workflow.

### Release note

```release-note
A sample plugin with E2E test cases has been added to demonstrate the usage of the CLI E2E framework API. This addition is designed to help plugin developers quickly get started with the CLI E2E framework API.

```

### CODE REVIEW

File: test/sample-plugin/.markdownlint.json
Comment (anujc25) [inline test/sample-plugin/.markdownlint.json:1]: Can we remove this file? Not sure we should include this for testing a sample plugin in Tanzu CLI repository.

File: .github/workflows/cli_core_e2e_sample-plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Tanzu CLI Sample-Plugin E2E Tests
```
Comment (anujc25) [inline .github/workflows/cli_core_e2e_sample-plugin.yaml:1]: How about calling this `Tanzu CLI Test Framework E2E Tests`?

**Review by anujc25 — COMMENTED** [review]
Thanks @chandrareddyp for adding this. Overall looks great. Few minor comments.

File: .github/workflows/cli_core_e2e_sample-plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  push:
+    branches: [ main, test*, release-*]
+
+jobs:
+  build:
+    name: Tanzu CLI Sample-Plugin E2E Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (anujc25) [inline .github/workflows/cli_core_e2e_sample-plugin.yaml:15]: I am not sure we need to run these tests on all OS Arch. If there is no good reason, we can just run these on `ubuntu` for now.

Comment (chandrareddyp) [reply .github/workflows/cli_core_e2e_sample-plugin.yaml]: updated to run only on ubuntu.

**Review by anujc25 — APPROVED** [review]
LGTM.

**Pull request merged**